### PR TITLE
Dev: fix test infra for GKE

### PR DESCRIPTION
**Pull request by pdecat**

### CODE REVIEW

**Comment by alexsomesan** [comment]
These exported env vars were recently removed in favor of the rendered kubeconfig file.
Do they work for you where kubeconfig doesn't?

**Comment by alexsomesan** [comment]
FYI: https://github.com/terraform-providers/terraform-provider-kubernetes/pull/259

**Comment by pdecat** [comment]
Let me check. That probably means only the README needs updating.

**Comment by alexsomesan** [comment]
The readme being stale is my fault :(

File: kubernetes/test-infra/gke/README.md [hunk 1]
```diff
@@ -20,11 +20,6 @@ See https://cloud.google.com/kubernetes-engine/versioning-and-upgrades#versions_
...
 ```
-export KUBE_HOST=https://$(terraform output endpoint)
-export KUBE_USER=$(terraform output username)
-export KUBE_PASSWORD=$(terraform output password)
-export KUBE_CLIENT_CERT_DATA="$(terraform output client_certificate_b64 | base64 -d -)"
-export KUBE_CLIENT_KEY_DATA="$(terraform output client_key_b64 | base64 -d -)"
-export KUBE_CLUSTER_CA_CERT_DATA="$(terraform output cluster_ca_certificate_b64 | base64 -d -)"
-export GOOGLE_ZONE=$(terraform output zone)
+export KUBECONFIG="${KUBECONFIG}:$(terraform output kubeconfig_path)"
+export GOOGLE_ZONE=$(terraform output google_zone)
```
Comment (alexsomesan) [inline kubernetes/test-infra/gke/README.md]: This one is supposed to be an input to the templates, so most of the time it's already set. But let's keep it in here for clarity and cases where a default is used.

More importantly, KUBE_CTX might need to be set as well, to the value of the relevant context from Kubeconfig.
Would you mind adding an output for that one too?

Thanks for taking care of this.

**Comment by pdecat** [comment]
It's ok with:

```
# export KUBECONFIG="$(terraform output kubeconfig_path)"
```

but not with:

```
# export KUBECONFIG="${KUBECONFIG}:$(terraform output kubeconfig_path)"
```

as the latter falls back on the current user's kubeconfig, e.g. minikube in my case:

```
TESTARGS="-run '^TestAccKubernetesRole_'" make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test $(go list ./... |grep -v 'vendor') -v -run '^TestAccKubernetesRole_' -timeout 120m
?       github.com/terraform-providers/terraform-provider-kubernetes    [no test files]
=== RUN   TestAccKubernetesRole_basic
--- FAIL: TestAccKubernetesRole_basic (0.05s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
                * kubernetes_role.test: 1 error occurred:
                * kubernetes_role.test: Post http://localhost/apis/rbac.authorization.k8s.io/v1/namespaces/default/roles: dial tcp 127.0.0.1:80: connect: connection refused




=== RUN   TestAccKubernetesRole_importBasic
--- FAIL: TestAccKubernetesRole_importBasic (0.03s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
                * kubernetes_role.test: 1 error occurred:
                * kubernetes_role.test: Post http://localhost/apis/rbac.authorization.k8s.io/v1/namespaces/default/roles: dial tcp 127.0.0.1:80: connect: connection refused




=== RUN   TestAccKubernetesRole_generatedName
--- FAIL: TestAccKubernetesRole_generatedName (0.04s)
    testing.go:538: Step 0 error: Error applying: 1 error occurred:
                * kubernetes_role.test: 1 error occurred:
                * kubernetes_role.test: Post http://localhost/apis/rbac.authorization.k8s.io/v1/namespaces/default/roles: dial tcp 127.0.0.1:80: connect: connection refused




FAIL
FAIL    github.com/terraform-providers/terraform-provider-kubernetes/kubernetes 0.144s
make: *** [GNUmakefile:17: testacc] Error 1
```

README updated accordingly.

File: kubernetes/test-infra/gke/README.md (hunk 1 above)
Comment (pdecat) [reply kubernetes/test-infra/gke/README.md]: Hmm, the template is using a datasource right now: `${data.google_compute_zones.available.names[0]}`: https://github.com/terraform-providers/terraform-provider-kubernetes/blob/v1.5.0/kubernetes/test-infra/gke/main.tf#L24

Using `export KUBECONFIG="$(terraform output kubeconfig_path)"` avoids having to set `KUBE_CTX` as there's only a single cluster and context in that file.

**Comment by alexsomesan** [comment]
I can reproduce that behaviour with two files in the KUBECONFIG variable.
It should work with more than one config - other clients do.
Looks like it's not picking up the correct context when more configurations are present.

Let's go for the change you proposed while we figure out where the client initialization is broken.

**Pull request merged**